src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -441,6 +441,15 @@ static StoragePtr create(const StorageFactory::Arguments & args)
     }
     else if (merging_params.mode == MergeTreeData::MergingParams::Replacing)
     {
+        /// Due to a misfortune, there could be an extra obsolete parameter.
+        /// We ignore it for backward compatibility.
+        if (arg_cnt - arg_num == 2 && !engine_args[arg_cnt - 1]->as<ASTLiteral>() && is_extended_storage_def)
```

Comment:
Silently ignoring it is even worse than having the server failing to start due to incompatibility